src/sentry/receivers/onboarding.py

```diff
@@ -539,11 +540,16 @@ def record_issue_tracker_used(plugin, project, user, **kwargs):
 
 @integration_added.connect(weak=False)
 def record_integration_added(
-    integration: Integration, organization: RpcOrganization, user, **kwargs
+    integration_id: int, organization_id: int, user_id: int | None, **kwargs
 ):
-    # TODO(Leander): This function must be executed on region after being prompted by control
+    integration: RpcIntegration | None = integration_service.get_integration(
+        integration_id=integration_id
+    )
+    if integration is None:
+        return
+
     task = OrganizationOnboardingTask.objects.filter(
-        organization_id=organization.id,
+        organization_id=organization_id,
         task=OnboardingTask.INTEGRATIONS,
     ).first()
```

Comment:
Should we aim to reduce our usage of signals? They don't lend themselves to readable easy to understand code and I'm concerned about how a generalized approach to sending signals between regions is going to age and perform when we are running multi-region for real.

Should we instead be adapting this kind of logic towards special cased outbox messages? Making operations that involve co-ordination between the region and control silo should be non-trivial to encourage us to find alternate solutions that don't make a tangle of network operations and eventual consistency.

Reply:
Talked this over in standup.